plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessageFactory.java

```diff
@@ -23,22 +24,9 @@ public static ErrorMessage createErrorMessage(Throwable e, int status) {
         int st = status;
         if (t instanceof OpenSearchException) {
             st = ((OpenSearchException) t).status().getStatus();
-        } else {
-            t = unwrapCause(e);
         }
+        t = MLExceptionUtils.getRootCause(t);
```

Comment:
would recommend to use `ExceptionsHelper.unwrapCause()` which provided by core library

https://github.com/opensearch-project/OpenSearch/blob/3fe66747b1523c74e06bc3465ef2ab682048f76b/libs/core/src/main/java/org/opensearch/ExceptionsHelper.java#L127-L145

Reply:
+1

Reply:
MLExceptionUtils.getRootCause(t) wraps the above function. What I was thinking is the existing class MLExceptionUtils is responsible for hosting all exception utilities in ml-commons, so I just simply wrap the above core function into this class to use for the entire ml-commons plugin.

Reply:
Also it can provide flexibility for ml-commons plugin. If we needed to change the logic, we only needed to change the function MLExceptionUtils.getRootCause(t), not everywhere the core function used.

Reply:
> MLExceptionUtils.getRootCause(t) wraps the above function 

I think the logic between `ExceptionUtils.getRootCause(t)` and `ExceptionsHelper.unwrapCause()` is different. `getRootCause` is to get the root cause but `unwrapCause` used for unwrap `OpenSearchWrapperException` and not go deeper. They may produce different result.

Reply:
Hi Hailong, I went through those two implementations and didn't see big differences. The only difference is about `ExceptionsHelper.unwrapCause()` having 10 times limitation on the extracting loop.

Reply:
> The only difference is about ExceptionsHelper.unwrapCause() having 10 times limitation on the extracting loop.

I guess the difference is that ExceptionsHelper.unwrapCause() will stop if the cause is not instance of `OpenSearchWrapperException`. I don't know if there has a case the cause is not `OpenSearchWrapperException` and still have inner cause.

Reply:
`OpenSearchWrapperException` is reusing the standard `throwable` `getCause` interface to unwrap. I prefer to use the method from the library `org.apache.commons.lang3.exception` which covers the method `ExceptionsHelper.unwrapCause()`.